index.js

```diff
@@ -11,6 +11,18 @@ const changePrototype = (to, from) => {
 	Object.setPrototypeOf(to, fromPrototype);
 };
 
+const wrappedToString = (withName, fromBody) => `/* Wrapped ${withName}*/\n${fromBody}`;
+
+const toStringName = Object.getOwnPropertyDescriptor(Function.prototype.toString, 'name');
+const toStringDescriptors = Object.getOwnPropertyDescriptor(Function.prototype, 'toString');
+
+const changeToString = (to, from, name) => {
+	const withName = name === '' ? '' : `with ${name.trim()}() `;
+	const newToString = wrappedToString.bind(null, withName, from.toString());
```

Comment:
We call `from.toString()` early (during `mimicFn()`) as opposed to lazily (during `to.toString()`). This is to ensure `from` can be garbage collected. Otherwise `to.toString()` closure would contain a reference to `from`.

We use `bind()` instead of a closure for the same reason. I'm not sure whether garbage collectors are keeping references of every variables in the closure scope (`changeToString()`), or only the ones referenced in the closure function (`wrappedToString()`), so I chose the safe side.

Also calling `from.toString()` early allows caching it in case `to.toString()` is called several times. I'm not sure whether that's already done by V8 under the hood though.

Reply:
I think some of this could be useful as code comments, for posterity.